StorageAdapter/ValuesTest.cs

```diff
@@ -0,0 +1,190 @@
+            {
+                Data = $"testData:{itemSuffix}"
+            };
+
+            // Act place item with initial data
+            putItemByKey(requestUri, contentToSet);
+
+            var returnedValue = getItemByKey(collectionId, itemKey);
+            Assert.Equal($"testData:{itemSuffix}", returnedValue.Data);
+
+            // Set item with updated data
+            contentToSet.ETag = returnedValue.ETag;
+            contentToSet.Data = "updatedData";
+            putItemByKey(requestUri, contentToSet);
+
+            // Assert that the value for this key was updated.
+            returnedValue = getItemByKey(collectionId, itemKey);
+            Assert.Equal($"updatedData", returnedValue.Data);
+        }
+
+        /// <summary>
+        /// Integration test verifying an item can be added and then deleted by key.
+        /// </summary>
+        [Fact, Trait(Constants.TEST, Constants.INTEGRATION_TEST)]
+        public void TheServiceCanDeleteItemByKey()
+        {
+            var itemSuffix = Guid.NewGuid().ToString();
+            var itemKey = "testKey";
+            var collectionId = "testCollection" + itemSuffix;
+
+            var contentToSet = new ValueApiModel()
+            {
+                Data = $"testData:{itemSuffix}"
+            };
+
+            var putRequest = new HttpRequest(Constants.STORAGE_ADAPTER_ADDRESS + $"/collections/{collectionId}/values/{itemKey}");
+            putRequest.SetContent(contentToSet);
+
+            var putResponse = this.httpClient.PutAsync(putRequest).Result;
+            Assert.Equal(HttpStatusCode.OK, putResponse.StatusCode);
+
+            // Delete the item using the same key.
+            var deleteRequest = new HttpRequest(Constants.STORAGE_ADAPTER_ADDRESS + $"/collections/{collectionId}/values/{itemKey}");
+            var deleteResponse = this.httpClient.DeleteAsync(putRequest).Result;
+            Assert.Equal(HttpStatusCode.OK, deleteResponse.StatusCode);
+
+            // Retrieve value by key and verify test item does not exist.
+            var getRequest = new HttpRequest(Constants.STORAGE_ADAPTER_ADDRESS + $"/collections/{collectionId}/values/{itemKey}");
+            var getResponse = this.httpClient.GetAsync(getRequest).Result;
+            Assert.Equal(HttpStatusCode.NotFound, getResponse.StatusCode);
+        }
+
+        /// <summary>
+        /// Integration test verifying an empty item leads to a bad request i.e. Status Code 400
+        /// </summary>
+        [Fact]
+        public void TheServiceReturnsBadRequestForEmptyContent()
+        {
+            var emptyPutRequest = new HttpRequest(Constants.STORAGE_ADAPTER_ADDRESS + $"/collections/emptyCollection/values");
+            emptyPutRequest.SetContent("");
+            var putResponse = this.httpClient.PostAsync(emptyPutRequest).Result;
+            Assert.Equal(HttpStatusCode.BadRequest, putResponse.StatusCode);
+        }
+
+        private void putItemByKey(string path, ValueApiModel content)
```

Comment:
nit method name should follow PascalCase

Reply:
Changed to pascal case for all methods.